src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -96,12 +104,29 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             {
                 typeSyntax = typeSyntax.StripRefIfNeeded();
 
+                var newTypeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+
+                if (newTypeSymbol.NullableAnnotation == NullableAnnotation.Annotated && variableIdentifier.HasValue)
+                {
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+                    // It's possible that the var shouldn't be annotated nullable, check assignments to the variable and 
+                    // determine if it needs to be null
+                    var methodBody = node.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsMethodBody);
```

Comment:
not certain how i feel about IsMethodBody.  What about things like code inside a lambda inside a field?

Reply:
what if this returns null?

we need tests for cases where not in a method.  esp cases like top-level statemetns, or being in a lambda in a field.